### ISSUE TITLE
[Integration] Request to support FAISSDocumentStore in 2.x

### ISSUE THREAD

**Issue by v3ss0n** [deepset-ai/haystack-core-integrations#717]
**Is your feature request related to a problem? Please describe.**
We are having problem migrating current version to 2.0 due to no FAISSSupport
**Describe the solution you'd like**
Due to is simplicity , performance and developer experience , we and many other RAG solution provider are now using FAISS . So it would be nice to integrate it.

**Describe alternatives you've considered**
QDrant and many others are not approved by some organization yet because they are very new and not backed by big vendors.

**Comment by masci** [comment]
Hi @v3ss0n and thanks for the feature request.

Do you envision an exact porting of the FAISS document store from Haystack 1.x, with the accompanying persistent storage, or would an in-memory, pure-FAISS solution also work?

**Comment by v3ss0n** [comment]
Exact porting is not necessary as long as we can use Fiass as DS , and yes of-course we are looking forward for both persistant and and in-memory DS but whatever work first.

**Comment by masci** [comment]
I think having in-memory, basic support for FAISS can be done rather easily. 
The problem with persistence is that FAISS alone can't manage metadata storage, so we need to pair it with another storage. We used sqlite in the previous version of Haystack and that turned out to be a mess, so this time it'd be better to pick a storage that's closer to the nature of FAISS - but this will require to prioritize this issue, and not sure this can happen soon.

**Comment by v3ss0n** [comment]
yeah i agree , sqlite was a mess and hard to manage. Lets start with in-membory.

**Comment by mlokhandwala** [comment]
Hi FAISS is more important then you might give it credit for, it is necessary for Haystack to move FAISS support to version 2. One of the most important attributes I like about FAISS That it uses local files and for document store, it can use SQL lite, This is a great advantage when we have small but many different isolated databases it may not be very scalable but there is a case to be made for small number of users requiring specialized RAG type applications.
